tests/unit/ControllerTest.php

```diff
@@ -177,4 +179,43 @@ public function testGetFormatIgnoresInvalid()
 
 		$this->assertEquals('cards', $result);
 	}
+
+	public function testDataUsesVar()
+	{
+		$file = fake(FileFaker::class);
+
+		$controller = new Files();
+		$controller->initController(service('request'), service('response'), service('logger'));
+
+		$method = $this->getPrivateMethodInvoker($controller, 'setData');
+		$method([
+			'files' => [
+				0 => $file
+			],
+		]);
+
+		$result = $controller->display();
+		$this->assertStringContainsString($file->filename, $result);
+	}
+
+	public function testDataUsesVarGotFileNotFound()
+	{
+		$file = fake(FileFaker::class);
+
+		$controller = new Files();
+		$controller->initController(service('request'), service('response'), service('logger'));
+
+		$method = $this->getPrivateMethodInvoker($controller, 'setData');
+		$method([
+			'files' => [
+				0 => (object) [
+					'filename' => 'foo.txt',
+					'thumbnail' => '',
+				]
+			],
+		]);
+
+		$this->expectException(FileNotFoundException::class);
```

Comment:
The controller should not throw unless it has been improperly configured by the developer. The only reason I can think why this is causing an exception might be because `Entities\File::locateDefaultThumbnail()` was not able to located the default? I haven't pulled the code to check it yet.

Reply:
@MGatner in controller, it seems need try catch to handle `FileNotFoundException` ?

```php
use CodeIgniter\Exceptions\PageNotFoundException;

try {

} catch (FileNotFoundException $e) {
     throw PageNotFoundException::forPageNotFound($e->getMessage());
}
```

Reply:
I want to find out where that exception is being thrown, because it is likely an underlying error.